### PR TITLE
Create migrating.md for SC4 to SC5 breaking changes

### PR DESCRIPTION
**Pull request by mfranzonello2**
Explicit callouts of what's changed from SC4 to SC5

### Description
Adding a page to Sauce Connect documentation calling out how to migrate from version 4 to version 5, specifically with what flags have changed or been removed, and what features like Selenium Relay that are no longer supported.

### Motivation and Context
Addresses confusion of what has changed between versions.

### Types of Changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation fix (typos, incorrect content, missing content, etc.)

### CODE REVIEW

**Comment by waggledans** [comment]
Should update the sidebar to include `SC4 Migration Guide` or smth like this, under Install

File: docs/secure-connections/sauce-connect-5/migrating.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+* Selenium Relay has been removed.
+* PID file has been removed.
+* Readyfile has been removed.
+* Log rotation has been removed.
+* Proxy autodetection is not supported.
+
+## Quick Start using `sc legacy`
+
+The `sc legacy` command is used to translate SC4 configuration to SC5.
```
Comment (waggledans) [inline docs/secure-connections/sauce-connect-5/migrating.md:36]: should add a link to `sc legacy` CLI. Can do it later though

File: docs/secure-connections/sauce-connect-5/migrating.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+| Version 4.x | Version 5.x |
+| ----------- | ----------- |
+| `--autodetect` | removed |
+| `--doctor` | removed |
+| `--experimental` | removed |
+| `--extra-info` | removed |
+| `--log-stats` | removed |
+| `--max-logsize` | removed |
+| `--no-autodetect` | removed |
+| `--no-remove-colliding-tunnels` | removed |
```
Comment (waggledans) [inline docs/secure-connections/sauce-connect-5/migrating.md:115]: `no-remove-colliding-tunnels` became `--tunnel-pool`

**Review by discombobulateme — APPROVED** [review]
Looks goos :)

**Pull request merged**